doc/database/multi_namespace_db_instances.md

```diff
@@ -0,0 +1,561 @@
+
+This approach is backward compatible to the Sonic devices with single NPU. The "globalDB" service could be considered equivalent to the database docker started in the linux "host" network namespace in single NPU device. We will discuss more on the design details of database docker service in this document.
+
+There will be a config_db.json file per database service, it will be named "/etc/sonic/config_db.json" for the "global DB" and /etc/sonic/config_db{NS}.json for the database service in the {NS} namespace.
+
+## Enhancements to Multi-DB design to support Multiple namespaces
+
+The file database_config.json contains the startup configuration which dictates the redis server host/port/unix_socket configurations + various available databases ( eg: APP_DB, CONFIG_DB ) etc. With the introduction of multiple namespaces, there is a need for separate database_config.json per namespace. This config file is created in the "working redis directories" during the database docker service startup. 
+
+The database service for a NPU linux namespace {NS} will use "/var/run/redis{NS}" as the "working redis directory" to create the various files like redis.sock, database_config.json etc. The "globalDB" database service would use "/var/run/redis" as the working directory.
+	
+![center](./img/redis.png)
+
+Following are the major design changes
+
+* A new file **databases.json** is introduced. It will contain the details of the namespaces and the corresponsing database_config.json files. This file would be created by the "globalDB" service in the directory "/var/run/redis/sonic-db/". In the below example, we consider a SONIC device with 3 NPUS's and hence have 3 namespaces refered with namespaceID "0", "1" & "2".
+
+```json
+[
+    {
+        "namespace" : "global",
+        "config" : "../redis/sonic-db/database_config.json"
+    },
+    {
+        "namespace" : "0",
+        "config" : "../redis0/sonic-db/database_config.json"
+    },
+        {
+        "namespace" : "1",
+        "config" : "../redis1/sonic-db/database_config.json"
+    },
+    {
+        "namespace" : "2",
+        "config" : "../redis2/sonic-db/database_config.json"
+    }
+]
+
+* The startup config file database_config.json is modified to have 'default' namespace DB instances + external references to the database instances present (if any) in database dockers running in other namesapces. There are a few attributes introduced, 
+
+	* "NAMESPACE" will store the namespace to which this database_config.json belongs.
+	* "INCLUDES" to point to the path of the databases.json file.
+
+* The database_config.json file for the "globalDB" service will have the "INSTANCES"/"DATABASES" (global DB service to have only APP_DB and CONFIG_DB) running localy in globalDB database docker on linux host namespace + INCLUDES attribute to refer to the external DB instances in the database docker services running in other NPU namespaces. The sample format is as below.
+
+```json
+{
+    "NAMESPACE": "global",
```

Comment:
>"NAMESPACE": "global", [](start = 4, length = 22)

Why we need this? If we can remove it, keep original version.

Reply:
The new file **databases.json** is a static file containing the namespace names and the correspoding database_config.json file.

The way I parse the database_config.json file (in class SonicDBConfig), is load the INSTANCES and DATABASES defined directly in the file as **SonicDBConfig._sonic_db_config['default'] = json.load(read_file) ** as default DB's
and then loop through all the files defined in the **"INCLUDES" : "../redis/sonic-db/databases.json",**. 

So when I loop through the files in **"../redis/sonic-db/databases.json"** I should skip the current NAMESPACE. I added this new attribute, to tell what namespace this "database_config.json" belongs to - and it is explicit detail which I felt was better.

Reply:
Removed the references to the keyword "namespace" in the json files.